File: leetcode_2490/circularSentence.java
package leetcode_2490;

class circularSentence{
    public boolean isCircularSentence(String sentence) {
        int len = sentence.length();
        for(int i=0; i<len; i++)
        {
            if(sentence.charAt(i) == ' ' && sentence.charAt(i-1) != sentence.charAt(i+1))
                return false;
        }
        return sentence.charAt(0) == sentence.charAt(len-1);
    }

    public static void main(String[] args) {
        String sentence = "leetcode exercises sound delightful";
        circularSentence obj = new circularSentence();
        System.out.println(obj.isCircularSentence(sentence));
    }
}

File: leetcode_2490/circularSentence.py
class circularSentence:
    def isCircularSentence(self, sentence: str) -> bool:
        # # Spliting method 
        # words = sentence.split(" ")
        
        # # This method works because it wont have an index out of range error
        # for i in range(len(words)):
        #     # current words at index i of the first letter is not equal to the previous word's last letter
        #     if words[i][0] != words[i-1][-1]: 
        #         return False
        # return True
        
        # One liner method
        for i in range(len(sentence)):
            if sentence[i] == " " and sentence[i-1] != sentence[i+1]:
                return False
        # Check if the first and last letter of the sentence are the same
        return sentence[0] == sentence[-1]

def main():
    sentence = "leetcode exercises sound delightful"
    obj = circularSentence()
    print(obj.isCircularSentence(sentence))

if __name__ == "__main__":
    main()
